lib/improver/spotdata/extract_data.py

```diff
@@ -0,0 +1,374 @@
+        for use in constructing new cubes.
+
+        """
+        indices = DimCoord(np.arange(len(latitudes)), long_name='index',
+                           units='1')
+        bd_ids = AuxCoord(site_ids, long_name='bestdata_id', units='1')
+        latitude = AuxCoord(latitudes, standard_name='latitude',
+                            units='degrees')
+        longitude = AuxCoord(longitudes, standard_name='longitude',
+                             units='degrees')
+        utc_offset = AuxCoord(utc_offsets, long_name='utc_offset',
+                              units='hours')
+        return indices, bd_ids, latitude, longitude, utc_offset
+
+    def make_cube(self, cube, data, sites):
+        """
+        Construct and return a cube containing the data extracted from the
+        grids by the desired method for the sites provided.
+
+        Args:
+        -----
+        cube : iris.cube.Cube
+            The original diagnostic cube from which data has been extracted.
+
+        data : numpy.array
+            Array of diagnostic values extracted for the defined sites.
+
+        sites : dict
+            A dictionary containing the properties of spotdata sites.
+
+        Returns:
+        --------
+        cube : iris.cube.Cube
+            An irregularly (i.e. non-gridded) cube of diagnostic data extracted
+            at the spotdata sites.
+
+        """
+        latitudes = [site['latitude'] for site in sites.itervalues()]
+        longitudes = [site['longitude'] for site in sites.itervalues()]
+        utc_offsets = [site['utc_offset'] for site in sites.itervalues()]
+        site_ids = sites.keys()
+
+        indices, bd_ids, latitude, longitude, utc_offset = (
+            self._build_coordinates(
+                latitudes, longitudes, site_ids, utc_offsets))
+
+        # Add leading dimension for time.
+        data.resize(1, len(data))
+        result_cube = Cube(data,
+                           long_name=cube.name(),
+                           dim_coords_and_dims=[(cube.coord('time'), 0),
+                                                (indices, 1)],
+                           aux_coords_and_dims=[(latitude, 1),
+                                                (longitude, 1),
+                                                (utc_offset, 1),
+                                                (bd_ids, 1)],
+                           units=cube.units)
+
+        # Enables use of long_name above for any name, and then moves it
+        # to a standard name if possible.
+        result_cube.rename(cube.name())
+        return result_cube
+
+    def use_nearest(self, cube, sites, neighbours):
+        """
+        Simplest case, in which the diagnostic data value at the nearest grid
+        point, as determined by the chosen PointSelection method, is used for
+        the site.
+
+        Args:
+        -----
+        See process() above.
+
+        """
+        if (not cube.coord_dims(cube.coord(axis='y').name())[0] == 0 or
```

Comment:
Maybe I'm misunderstanding this, but why are you using "not" and "==", rather than "!="?